arcade/gui/property.py

```diff
@@ -18,21 +18,39 @@ class _Obs(Generic[P]):
     def __init__(self, value: P):
         self.value = value
         # This will keep any added listener even if it is not referenced anymore and would be garbage collected
-        self.listeners: Set[Callable[[], Any]] = set()
+        self.listeners: Set[Callable[[Any, Any], Any]] = set()
```

Comment:
Can we make this a generic somehow? As a warning, although I haven't used `ParamSpec` extensively, I know it can be limited in some ways if you use either older versions of the `typing_extensions` backports.

Reply:
the first arg can be what ever instance the property is defined in, the second is the new value, which could be resolved by Property[P] i guess.